lib/api/filter.rb

```diff
@@ -82,6 +90,12 @@ def parse_filter(filter)
       str_method = is_regex ? methods[:regex] : methods[:default]
 
       filter_value, method = case filter_value
+                             when /^\[(.*)\]$/
+                               unless methods[:array]
+                                 raise BadRequestError, "Unsupported operator for arrays: #{operator}"
+                               end
+
+                               [$1.split(","), methods[:array]]
```

Comment:
I like how `unquoted_filter_value` introduced a variable close to the regular expression. So it is very clear the `$1` reference.

Although, with that case, it is nicer because the value can be used asis.

Can you do something similar here?

`array_contents = $1.split(",")`

Reply:
Yeah, I am fine with that idea.  Was going with terse, but I can be a bit verbose if you prefer.